### PR TITLE
Add write capability to PilzModbusReadClient

### PR DESCRIPTION
**Pull request by hslusarek**

### CODE REVIEW

File: prbt_hardware_support/include/prbt_hardware_support/libmodbus_client.h [hunk 1]
```diff
@@ -30,50 +30,22 @@ namespace prbt_hardware_support
...
  */
 class LibModbusClient : public ModbusClient
 {
 public:
+  virtual ~LibModbusClient() override;
 
-  virtual ~LibModbusClient();
+  bool init(const char* ip, unsigned int port) override;
 
-  /**
```
Comment (ct2034) [inline prbt_hardware_support/include/prbt_hardware_support/libmodbus_client.h:41]: Please add all the documentation again

File: prbt_hardware_support/include/prbt_hardware_support/param_names.h [hunk 2]
```diff
@@ -27,6 +27,7 @@ static const std::string PARAM_MODBUS_SERVER_IP_STR {"modbus_server_ip"};
 static const std::string PARAM_MODBUS_SERVER_PORT_STR {"modbus_server_port"};
 static const std::string PARAM_MODBUS_RESPONSE_TIMEOUT_STR {"modbus_response_timeout"};
 static const std::string PARAM_MODBUS_TOPIC_NAME_STR {"modbus_topic_name"};
```
Comment (ct2034) [inline prbt_hardware_support/include/prbt_hardware_support/param_names.h]: rename this to PARAM_MODBUS_READ_TOPIC_NAME_STR
Not sure about  the topic name itself. maybe it should stay like this, for compatibility. Maybe do a search and replace in the repo to do modbus_topic_name -> modbus_read_topic_name"

File: prbt_hardware_support/include/prbt_hardware_support/pilz_modbus_client_exception.h [hunk 3]
```diff
@@ -23,16 +23,16 @@
...
+      PilzModbusClientException( const std::string& what_arg ):
         std::runtime_error(what_arg)
       {
       }
   };
 }
 
-#endif // PRBT_HARDWARE_SUPPORT_READ_CLIENT_EXCEPTION_H
\ No newline at end of file
+#endif // PRBT_HARDWARE_SUPPORT_READ_CLIENT_EXCEPTION_H
```
Comment (ct2034) [inline prbt_hardware_support/include/prbt_hardware_support/pilz_modbus_client_exception.h]: change the precompiler variable PRBT_HARDWARE_SUPPORT_READ_CLIENT_EXCEPTION_H -> PRBT_HARDWARE_SUPPORT_MODBUS_CLIENT_EXCEPTION_H

File: prbt_hardware_support/src/pilz_modbus_client_node.cpp [hunk 4]
```diff
@@ -104,15 +104,19 @@ int main(int argc, char **argv)
...
+  prbt_hardware_support::PilzModbusClient modbus_client(nh,
                                                             static_cast<uint32_t>(num_registers_to_read),
                                                             static_cast<uint32_t>(index_of_first_register),
                                                             std::unique_ptr<LibModbusClient>(new LibModbusClient()),
                                                             static_cast<unsigned int>(response_timeout_ms),
-                                                            modbus_topic_name);
+                                                            modbus_topic_name, modbus_write_topic_name);
 
   ROS_DEBUG_STREAM("Modbus client IP: " << ip << " | Port: " << port);
   ROS_DEBUG_STREAM("Number of registers to read: " << num_registers_to_read
```
Comment (ct2034) [inline prbt_hardware_support/src/pilz_modbus_client_node.cpp:89]: are they only to read? or also to write?
if 1) add debug output for write
if 2) change variable name

File: prbt_hardware_support/test/include/prbt_hardware_support/pilz_modbus_client_mock.h [hunk 5]
```diff
@@ -33,7 +33,10 @@ class PilzModbusClientMock : public prbt_hardware_support::ModbusClient
   MOCK_METHOD0(connect, void());
   MOCK_METHOD1(setResponseTimeoutInMs, void(unsigned int timeout_ms));
   MOCK_METHOD2(readHoldingRegister, std::vector<uint16_t>(int addr, int nb));
+  MOCK_METHOD4(writeReadHoldingRegister, std::vector<uint16_t>(const int write_addr,
+                                                               const std::vector<uint16_t>& write_reg,
+                                                               const int read_addr, const int read_nb));
```
Comment (ct2034) [inline prbt_hardware_support/test/include/prbt_hardware_support/pilz_modbus_client_mock.h]: add newline between arguments to match style

File: prbt_hardware_support/src/pilz_modbus_client_node.cpp [hunk 6]
```diff
@@ -66,7 +66,7 @@ T getParam(const ros::NodeHandle& nh, const std::string& param_name)
 using namespace prbt_hardware_support;
 
 /**
- * @brief Read requested parameters, start and initialize the prbt_hardware_support::PilzModbusReadClient
+ * @brief Read requested parameters, start and initialize the prbt_hardware_support::PilzModbusClient
  */
 int main(int argc, char **argv)
 {
```
Comment (ct2034) [inline prbt_hardware_support/src/pilz_modbus_client_node.cpp:39]: ros::init(argc, argv, "modbus_read_node");
->   ros::init(argc, argv, "modbus_client_node"); (or whatever)

File: prbt_hardware_support/include/prbt_hardware_support/modbus_topic_definitions.h [hunk 7]
```diff
@@ -24,6 +24,7 @@ namespace prbt_hardware_support
 {
 // Topic names
 static const std::string TOPIC_MODBUS_READ = "/pilz_modbus_node/modbus_read";
+static const std::string TOPIC_MODBUS_WRITE = "/pilz_modbus_node/modbus_write";
```
Comment (ct2034) [inline prbt_hardware_support/include/prbt_hardware_support/modbus_topic_definitions.h]: update documentation in prbt_hardware_support/README.md

**Review by ct2034 — CHANGES_REQUESTED** [review]

**Comment by hslusarek** [comment]
@ct2034 Please do not force push on branches on which other people work on! If you want to change something, then do a normal commit and push it. Otherwise, synchronize yourself with the people which work on the branches.

**Pull request merged**